hot-reload.md

```diff
@@ -47,7 +46,7 @@ A code change has a visible effect only if the modified Dart code is run again
 after the change. The following section describes common code patterns where the
 modified code will _not_ run again after hot reload.
 
-## Troubleshooting hot reload
+## Understanding when a full app restart is needed
```

Comment:
@a-siva @rmacnak-google  This set of changes also changes the meaning. We discussed how to work around the edge cases, right? Is this now the same as the last section "Understanding limitations"?

What point do we want to make? I'm perfectly willing to research whether it's possible to reinitialize in IntelliJ... if that's the direction you want to go in.

Reply:
Yes, I intentionally changed the meaning.

'Troubleshooting' is something you do when something is broken and doesn't have it's intended function. That is not the issue here. Hot reload behaves as designed, and for good reasons.

Reply:
We could change the header to "Refactoring to enable hot reload"... WDYT?

Re: the meaning of the word "troubleshooting", this is from the user's perspective which is different from ours. The content is about hot reload not behaving the way a user was expecting it to, and they're going to make small code changes to remedy that. The question of whether or not they're in this situation due to team design decisions and/or technical constraints, that mixes our perspective in where it's not relevant. Their goal is to enable hot reload in as many situations as possible...

The following section, "Understanding limitations", is about full app restart. That's when no code changes can help.

Reply:
In the spirit of landing this PR, I have reverted this change.

I'll be in MTV next week, then we can spend some together on this topic.